packages/grid/src/vaadin-grid-a11y-mixin.js

```diff
@@ -42,16 +42,20 @@ export const A11yMixin = (superClass) =>
 
     /** @protected */
     _a11yUpdateHeaderRows() {
-      Array.from(this.$.header.children).forEach((headerRow, index) =>
-        headerRow.setAttribute('aria-rowindex', index + 1),
-      );
+      Array.from(this.$.header.children).forEach((headerRow, index, array) => {
+        headerRow.setAttribute('aria-rowindex', index + 1);
+        this._updateRowBodyCellsPart(headerRow, 'first-header-row-cell', index === array.length - 1);
```

Comment:
```suggestion
        this._updateRowBodyCellsPart(headerRow, 'first-header-row-cell', index === 0);
```
Should check for the first row I think.

A bit confusing that this calls `_updateRowBodyCellsPart` with a header row.